tests/js/server/dump/dump-move-shard.js

```diff
@@ -0,0 +1,126 @@
+/* jshint strict: false, sub: true */
+/* global print, arango, assertEqual, assertTrue, assertFalse */
+'use strict';
+
+// /////////////////////////////////////////////////////////////////////////////
+// DISCLAIMER
+//
+// Copyright 2016-2019 ArangoDB GmbH, Cologne, Germany
+// Copyright 2014 triagens GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// @author Kaveh Vahedipour
+// //////////////////////////////////////////////////////////////////////////////
+
+var internal = require("internal");
+var jsunity = require("jsunity");
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function dumpTestSuite () {
+  'use strict';
+  var db = internal.db;
+
+  return {
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test the empty collection
+////////////////////////////////////////////////////////////////////////////////
+
+    testMoveShard : function () {
+
+      if (arango.getRole() !== "COORDINATOR") {
+        return;
+      }
+
+      let database = "UnitTestsDumpDst";
+      let collection = "UnitTestsDumpReplicationFactor1";
+      let planShards =
+          arango.GET("_admin/cluster/shardDistribution").results[collection].Plan;
+
+      /*eslint no-extend-native: ["error", { "exceptions": ["Array"] }]*/
+      Array.prototype.remove = function() {
+        var what, a = arguments, L = a.length, ax;
+          while (L && this.length) {
+          what = a[--L];
+          while ((ax = this.indexOf(what)) !== -1) {
+            this.splice(ax, 1);
+          }
+        }
+        return this;
+      };
+
+      var i = 0;
+      var pending = [];
+      Object.keys(planShards).forEach(
+        function (shard) {
+          let dbs = ["DBServer0001", "DBServer0002", "DBServer0003"];
```

Comment:
Is it guaranteed by our tests that servers will be named like this? I can't say for sure, but potentially we should rather use `require("@arangodb/test-helper").getServersByType("dbserver")` here.

Reply:
Yes, the names are guaranteed in this setting.